src/Sarif/ExtensionMethods.cs

```diff
@@ -514,6 +514,45 @@
             return propertyBag;
         }
 
+        public static IEnumerable<T> MergeWithList<T>(
```

Comment:
>MergeWithList [](http://example.com/codeflow?start=37&length=13)

This is a common method, with the current name,
This method not just do merge but also do distinct, 
if original list has 3 items and the second list contains 0 item, 
the result returns only 1 item, the person calling this method will be surprised. 

Not sure if there is a way to make the name clear 
e.g. 
`DistictMergeWithList`

Reply:
updated